source/slang/core.meta.slang

```diff
@@ -1720,6 +1720,57 @@ T __attachToNativeRef(NativeRef<T> nativeVal)
 }
 
 //@public:
+__generic<T, let N:int>
+__magic_type(ArrayExpressionType)
+struct Array : IRWArray<T>
+{
+    __intrinsic_op($(kIROp_GetArrayLength))
+    int getCount();
+}
+
+//@public:
+__generic<int N>
+__magic_type(StringLiteralExpressionType)
+__intrinsic_type($(kIROp_StringLiteralType))
+struct StringLiteral : IArray<uint>
```

Comment:
Can we name this `ShortString`? We should also add documentation  to the type, something like "Represent a string value backed by statically allocated storage."

`StringLiteral` is a term that has special meaning in the type system context, and I prefer we don't use it here.